src/codec/vp8/bool_decoder.rs

```diff
@@ -50,19 +49,18 @@ pub enum BoolDecoderError {
 pub type BoolDecoderResult<T> = std::result::Result<T, BoolDecoderError>;
 
 /// The decoder state.
-#[derive(Default)]
-pub struct BoolDecoder<T> {
-    data: Cursor<T>,
+pub struct BoolDecoder<'a> {
+    data: NaluReader<'a>,
     range: usize,
     value: usize,
     count: isize,
 }
 
-impl<T: AsRef<[u8]>> BoolDecoder<T> {
+impl<'a> BoolDecoder<'a> {
     /// Creates a new instance.
-    pub fn new(data: T) -> Self {
+    pub fn new(data: &'a [u8]) -> Self {
         Self {
-            data: Cursor::new(data),
+            data: NaluReader::new(data, false),
```

Comment:
I think this is too confusing. It feels like there should be a "bit reader" if you want to read bits from a slice. You aren't using any of the NALU parsing features in NaluReader.